modules/table/src/main/java/org/apache/ignite/internal/table/KeyValueBinaryViewImpl.java

```diff
@@ -240,7 +273,7 @@ public boolean remove(@Nullable Transaction tx, @NotNull Tuple key, @NotNull Tup
         }
 
         return tbl.deleteAll(keyRows, (InternalTransaction) tx)
-                .thenApply(t -> t.stream().filter(Objects::nonNull).map(this::wrap).map(TableRow::valueTuple).collect(Collectors.toList()));
+                       .thenApply(t -> t.stream().filter(Objects::nonNull).map(this::unmarshal).collect(Collectors.toList()));
```

Comment:
Please, rewrite it with code that does not use Stream API. This code potentially is on a hot path.